javaslang/src/test/java/javaslang/collection/BitSetTest.java

```diff
@@ -0,0 +1,389 @@
+        return this.<Double>bsBuilder().ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
+    }
+
+    @Override
+    protected BitSet<Integer> rangeClosed(int from, int toInclusive) {
+        if (badRange(from, toInclusive)) {
+            return this.<Integer>bsBuilder().ofAll(Iterator.rangeClosed(from, toInclusive));
+        } else {
+            return BitSet.rangeClosed(from, toInclusive);
+        }
+    }
+
+    @Override
+    protected BitSet<Integer> rangeClosedBy(int from, int toInclusive, int step) {
+        if (badRange(from, toInclusive)) {
+            return this.<Integer>bsBuilder().ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
+        } else {
+            return BitSet.rangeClosedBy(from, toInclusive, step);
+        }
+    }
+
+    @Override
+    protected BitSet<Long> rangeClosed(long from, long toInclusive) {
+        if (badRange(from, toInclusive)) {
+            return this.<Long>bsBuilder().ofAll(Iterator.rangeClosed(from, toInclusive));
+        } else {
+            return BitSet.rangeClosed(from, toInclusive);
+        }
+    }
+
+    @Override
+    protected BitSet<Long> rangeClosedBy(long from, long toInclusive, long step) {
+        if (badRange(from, toInclusive)) {
+            return this.<Long>bsBuilder().ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
+        } else {
+            return BitSet.rangeClosedBy(from, toInclusive, step);
+        }
+    }
+
+    // BitSet specific
+
+    @Test
+    public void testBitSet1() {
+        BitSet<Integer> bs = BitSet.empty();
+
+        bs = bs.add(2);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.add(4);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.remove(2);
+        assertThat(bs.head()).isEqualTo(4);
+        assertThat(bs.contains(2)).isFalse();
+
+        bs = bs.remove(4);
+        assertThat(bs.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void testBitSet2() {
+        BitSet<Integer> bs = BitSet.empty();
+
+        bs = bs.add(2);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.add(70);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.remove(2);
+        assertThat(bs.head()).isEqualTo(70);
+        assertThat(bs.contains(2)).isFalse();
+
+        bs = bs.remove(70);
+        assertThat(bs.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void testBitSetN() {
+        BitSet<Integer> bs = BitSet.empty();
+
+        bs = bs.add(2);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.add(700);
+        assertThat(bs.head()).isEqualTo(2);
+
+        bs = bs.remove(2);
+        assertThat(bs.head()).isEqualTo(700);
+        assertThat(bs.contains(2)).isFalse();
+
+        bs = bs.remove(700);
+        assertThat(bs.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void testFactories() {
+        assertThat(BitSet.of(7).contains(7)).isTrue();     // BitSet1
```

Comment:
* < 64
* < 2*64
* >= 2*64

? we may include it in the comments or the values to make it obvious why the numbers were chosen